src/Microsoft.Diagnostics.Monitoring.WebApi/ParameterCapturing/CapturedParametersFormatter.cs

```diff
@@ -0,0 +1,116 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Diagnostics.Monitoring.Options;
+using Microsoft.Diagnostics.Monitoring.WebApi.Models;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostics.Monitoring.WebApi.ParameterCapturing
+{
+    internal static class CapturedParametersFormatter
+    {
+        public static Task WriteAsync(
+            IEnumerable<ICapturedParameters> parameters,
+            CapturedParameterFormat format,
+            Stream outputStream,
+            CancellationToken token)
+        {
+            return format switch
+            {
+                CapturedParameterFormat.PlainText => WriteParametersAsPlainText(parameters, outputStream, token),
+                CapturedParameterFormat.Json => WriteParametersAsJson(parameters, outputStream, token),
+                _ => throw new InvalidOperationException(),
+            };
+        }
+
+        private static Task WriteParametersAsJson(IEnumerable<ICapturedParameters> parameters, Stream outputStream, CancellationToken token)
+        {
+            CapturedParametersResult result = BuildResultModel(parameters);
+
+            return JsonSerializer.SerializeAsync(outputStream, result, cancellationToken: token);
+        }
+
+        private const string Indent = "  ";
+
+        private static async Task WriteParametersAsPlainText(IEnumerable<ICapturedParameters> parameters, Stream outputStream, CancellationToken token)
+        {
+            CapturedParametersResult result = BuildResultModel(parameters);
+
+            await using StreamWriter writer = new StreamWriter(outputStream, Encoding.UTF8, leaveOpen: true);
+            await writer.WriteLineAsync("Captured Parameters:");
+
+            StringBuilder builder = new();
+
+            foreach (CapturedMethod capture in result.CapturedMethods)
+            {
+                builder.AppendLine($"[{capture.CapturedDateTime}][thread {capture.ThreadId}] {GetValueOrUnknown(capture.ActivityId)}[format: {capture.ActivityIdFormat}]");
```

Comment:
All of these interpolated strings should be using `FormattableString.Invariant` so we don't have unexpected formatting from the current culture provider